sim/firesim-lib/src/main/cc/bridges/tracerv.cc

```diff
@@ -285,6 +296,10 @@ size_t tracerv_t::process_tokens(int num_beats, int minimum_batch_beats) {
 }
 
 void tracerv_t::tick() {
+  if (!init_ran) {
```

Comment:
This is an invariant which should be respected by all drivers and the message is not too relevant at runtime. It should be an assertion.

Reply:
I'm confused? It's not an invariant because the call of `init()` happens outside the constructor. The way firesim is setup, it's required that `init()` is called before any ticks. I added this because I personally ran into an issue where I forgot to call init and got weird behavior. I forgot because I'm still relatively a new user to firesim, and this is the first bridge test I've written after all of your changes. I look at it like guard rails for developers. I agree it should be moved higher, maybe into `bridge_driver_t` but I'm not sure how.

Reply:
Nandor's refactoring should make it very difficult to forget to call Init because the control flow is more prescribed. 

Perhaps a better way to do this would be to move it into a parent class (i.e, bridge_driver_t), so that this invariant is checked across all bridges. Maybe pull this change out of this PR and do a follow up? @nandor thoughts?

Reply:
I think we can do away with the check here completely. If `init` allocated some data or created some config vars I'd add an assertion, but there's really no point in adding a flag just for this.